From 65d261e40f03bf23fa694b429961e11d05523bef Mon Sep 17 00:00:00 2001
From: Tian Tao <tiantao6@hisilicon.com>
Date: Thu, 21 Jan 2021 17:57:10 +0800
Subject: [PATCH 13/13] mm/zswap: fix variable 'entry' is uninitialized when
 used

the entry has not been initialized when it is used, so allocate PAGE_SIZE

Signed-off-by: Tian Tao <tiantao6@hisilicon.com>
Reported-by: kernel test robot <lkp@intel.com>
Link: https://lore.kernel.org/r/1611223030-58346-1-git-send-email-tiantao6@hisilicon.com
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 mm/zswap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/zswap.c b/mm/zswap.c
index a85334b719b9..00789d404aef 100644
--- a/mm/zswap.c
+++ b/mm/zswap.c
@@ -944,7 +944,7 @@ static int zswap_writeback_entry(struct zpool *pool, unsigned long handle)
 
 	if (!zpool_can_sleep_mapped(pool)) {
 
-		tmp = kmalloc(entry->length, GFP_ATOMIC);
+		tmp = kmalloc(PAGE_SIZE, GFP_ATOMIC);
 		if (!tmp)
 			return -ENOMEM;
 	}
-- 
2.30.0

